README.adoc

```diff
@@ -0,0 +1,138 @@
+// vim: tw=79
+= Tendrl Node Agent
+
+Tendrl node agent resides on every node managed by tendlr. It is responsible
+for operating system level operations such as hardware inventory, service
+management, process monitoring etc. The node agent also serves as the
+provisioning controller and can invoke provisioning operations on the node.
+
+* Free software: LGPL2
+* Documentation: TODO (rk)
+* Source: https://github.com/Tendrl/node_agent
+* Bugs: https://github.com/Tendrl/node_agent/issues
+
+image:https://travis-ci.org/Tendrl/node_agent.svg?branch=master[Build status,
+link="https://travis-ci.org/Tendrl/node_agent"]
+image:https://coveralls.io/repos/github/Tendrl/node_agent/badge.svg?branch=master[Coverage,
+link="https://coveralls.io/github/Tendrl/node_agent?branch=master"]
+
+
+== Features
+
+TODO
+
+
+== Installation from Source on CentOS 7
+
+IMPORTANT: Node agent needs to be installed on every node of the storage
+cluster that is to be managed by tendrl.
+
+NOTE: All the commands are run as a regular user that has `sudo` privileges.
+The commands are all assumed to be run from a single directory, which by
+default could be the user's home directory.
+
+=== Deployment Requirements
+
+. Ensure that etcd is running on a node in the network and is reachable from
+the node you're about to install the node agent on. Note it's address and port.
+
+=== System Setup
+
+. Install the build toolchain and other development packages.
+
+ $ sudo yum groupinstall 'Development Tools'
+ $ sudo yum install libffi-devel openssl-devel
+
+. Install https://virtualenvwrapper.readthedocs.io/[virtualenvwrapper].
+
+ $ sudo yum install python-virtualenv python-virtualenvwrapper
+
+. Create system directories.
+
+ $ sudo mkdir -p /etc/tendrl /var/log/tendrl/node_agent
```

Comment:
The 'common' directory under /var/log/tendrl can exist for now, but it needs to be re-evaluated later. I've opened an issue to tackle this: https://github.com/Tendrl/common/issues/62